src/nagendra/JavaArray.java

```diff
@@ -0,0 +1,84 @@
+package nagendra;
+
+public class JavaArray {
+
+	int getMinimumNumber(int[] num) {
+
+		int minimum = num[0];
+
+		for (int index = 0; index < num.length; index++) {
+			if (minimum > num[index]) {
+				minimum = num[index];
+			}
+		}
+		return minimum;
+	}
+
+	int getMaximumNumber(int[] num) {
+
+		int maximum = num[0];
+
+		for (int index = 0; index < num.length; index++) {
+			if (maximum < num[index]) {
+				maximum = num[index];
+			}
+		}
+		return maximum;
+	}
+
+	int getAverageNumber(int[] num) {
+
+		int totalSum = 0;
+		int average;
+		for (int index = 0; index < num.length; index++) {
+			totalSum += num[index];
+		}
+		average = totalSum / num.length;
+		return average;
+	}
+
+	void displayAllNumbers(int[] num) {
+
+		System.out.println(" ---The Numbers from 0 till Array.length are --- ");
+
+		for (int index = 0; index < num.length; index++) {
+			System.out.println(" --->" + num[index]);
+		}
+	}
+
+	void displayReverseNumbers(int[] num) {
+
+		System.out.println(" ---The Numbers from Array.length till 0 are --- ");
+
+		for (int index = num.length - 1; index >= 0; index--) {
+			System.out.println(" --->" + num[index]);
+		}
+	}
+
+	void getAverageOfMinAndMax(int minimum, int maximum) {
+		int average = (minimum + maximum) / 2;
```

Comment:
I think we have requirement where you need to find max & min number then you need to find avg of both, so this expecting two variable in method looks wrong as per the implementation perspective, user will give us array, not min & max value. I hope you got my point.


You can write code like this, I remove variable and directly doing sum of min & max, no problem, if you are not comfortable to write code like below, just optimize.

int avgOfMinMax = (getMinNumber(input)+getMaxNumber(input))/2